src/items.rs

```diff
@@ -1892,25 +1911,33 @@ pub(crate) fn rewrite_struct_field(
     if prefix.is_empty() && !attrs_str.is_empty() && attrs_extendable && spacing.is_empty() {
         spacing.push(' ');
     }
-    let orig_ty = shape
+
+    // Question. Why don't we immediately return None/Err with ?
+    // let orig_ty = shape
+    //     .offset_left(overhead + spacing.len())
+    //     .and_then(|ty_shape| field.ty.rewrite(context, ty_shape));
+    let ty_shape = shape
         .offset_left(overhead + spacing.len())
-        .and_then(|ty_shape| field.ty.rewrite(context, ty_shape));
-    if let Some(ref ty) = orig_ty {
+        .max_width_error(shape.width, field.ty.span())?;
+    let orig_ty = field.ty.rewrite_result(context, ty_shape);
+    if let Ok(ref ty) = orig_ty {
         if !ty.contains('\n') && !contains_comment(context.snippet(missing_span)) {
-            return Some(attr_prefix + &spacing + ty);
+            return Ok(attr_prefix + &spacing + ty);
```

Comment:
We need to rework this. We can't fail fast by using the try operator when calculating `ty_shape`. In the original code when we failed to calculate the `orig_ty` shape we wouldn't enter the `if let Some(ref ty) = orig_ty` block, and code execution would continue on to the code below. The current implementation short circuits if we fail to calculate the `ty_shape` which changes the behavior.